prow/plugins/label/label.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package label
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/util/sets"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const pluginName = "label"
+
+var (
+	labelRegex       = regexp.MustCompile(`(?m)^/(area|priority|kind)\s*(.*)$`)
+	sigMatcher       = regexp.MustCompile(`(?m)@sig-([\w-]*)-(?:misc|test-failures|bugs|feature-requests|proposals|pr-reviews|api-reviews)`)
+	nonExistentLabel = "`%v` does not exist in this repository"
+	labelFailed      = "The following labels: %v could not be added to the issue or PR"
+)
+
+func init() {
+	plugins.RegisterIssueCommentHandler(pluginName, handleIssueComment)
+}
+
+type githubClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+	IsMember(org, user string) (bool, error)
+	AddLabel(owner, repo string, number int, label string) error
+	GetLabels(owner, repo string) ([]string, error)
+}
+
+func handleIssueComment(pc plugins.PluginClient, ic github.IssueCommentEvent) error {
+	return handle(pc.GitHubClient, ic)
+}
+
+func handle(gc githubClient, ic github.IssueCommentEvent) error {
+	commenter := ic.Comment.User.Login
+	owner := ic.Repo.Owner.Name
+	repo := ic.Repo.Name
+	number := ic.Issue.Number
+
+	labelMatches := labelRegex.FindAllStringSubmatch(ic.Comment.Body, -1)
+	sigMatches := sigMatcher.FindAllStringSubmatch(ic.Comment.Body, -1)
+	if len(labelMatches) == 0 && len(sigMatches) == 0 {
+		return nil
+	}
+
+	member, err := gc.IsMember(repo, commenter)
+	if err != nil {
+		return fmt.Errorf("IsMember failed: %v", err)
+	}
+	if !member {
+		return gc.CreateComment(owner, repo, number, plugins.FormatResponse(ic.Comment, "Only kubernetes org members can add labels."))
```

Comment:
As written, only `ic.Repo.Owner.Name` members can add labels, not k8s members. Prow also hits kubernetes-incubator among a few others.